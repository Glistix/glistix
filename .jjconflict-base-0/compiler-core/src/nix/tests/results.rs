use crate::assert_nix;

#[test]
fn ok() {
    assert_nix!(r#"pub fn main() { Ok(1) }"#);
}

#[test]
fn error() {
    assert_nix!(r#"pub fn main() { Error(1) }"#);
}

#[test]
fn ok_fn() {
    assert_nix!(r#"pub fn main() { Ok }"#);
}

#[test]
fn error_fn() {
    assert_nix!(r#"pub fn main() { Error }"#);
}

#[test]
fn qualified_ok() {
    assert_nix!(
        r#"import gleam
pub fn main() { gleam.Ok(1) }"#
    );
}

#[test]
fn qualified_error() {
    assert_nix!(
        r#"import gleam
pub fn main() { gleam.Error(1) }"#
    );
}

#[test]
fn qualified_ok_fn() {
    assert_nix!(
        r#"import gleam
pub fn main() { gleam.Ok }"#
    );
}

#[test]
fn qualified_error_fn() {
    assert_nix!(
        r#"import gleam
pub fn main() { gleam.Error }"#
    );
}

#[test]
fn aliased_ok() {
    assert_nix!(
        r#"import gleam.{Ok as Thing}
pub fn main() { Thing(1) }"#
    );
}

#[test]
fn aliased_error() {
    assert_nix!(
        r#"import gleam.{Error as Thing}
pub fn main() { Thing(1) }"#
    );
}

#[test]
fn aliased_ok_fn() {
    assert_nix!(
        r#"import gleam.{Ok as Thing}
pub fn main() { Thing }"#
    );
}

#[test]
fn aliased_error_fn() {
    assert_nix!(
        r#"import gleam.{Error as Thing}
pub fn main() { Thing }"#
    );
}
