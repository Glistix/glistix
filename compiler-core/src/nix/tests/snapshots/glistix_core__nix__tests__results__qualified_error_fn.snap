---
source: compiler-core/src/nix/tests/results.rs
expression: "import gleam\npub fn main() { gleam.Error }"
snapshot_kind: text
---
----- SOURCE CODE
import gleam
pub fn main() { gleam.Error }

----- COMPILED NIX
let
  gleam' = builtins.import ./../gleam.nix;
  
  main = { }: gleam'.Error;
in
{ inherit main; }
